### PR TITLE
Cu ut9pye initialize python project

### PR DESCRIPTION
**Pull request by jmercadoc**
Resolves [#ut9pye]

created denue django project
create api app
created models for denue schema
created tests
updated README

how to tests:
docker-compose exec denue-api python denue/manage.py test api.tests.tests_model_comercial_activity
docker-compose exec denue-api python denue/manage.py test api.tests.tests_model_stratum
docker-compose exec denue-api python denue/manage.py test api.tests.tests_model_shop_type
docker-compose exec denue-api python denue/manage.py test api.tests.tests_model_shop

### CODE REVIEW

**Review by maurogonzalez — COMMENTED** [review]
Mainly boilerplate. Better focus in the logic applying FP and testing (TDD as a suggestion).

**Review by maurogonzalez — COMMENTED** [review]
In the future, add portability, i.e., avoid modification of the state of the machine (ptyhon installs, versions, etc.).

**Review by maurogonzalez — APPROVED** [review]
Looks good!

File: denue/api/tests.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+
+from django.test import TestCase
+
+from api.models import ComercialActivity, Stratum, ShopType, Shop
+from fixtures.utilities import * 
+from .functions_test import *
+
+
+# Create your tests here.
+class ComercialActivityTestCase(TestCase):
```
Comment (maurogonzalez) [inline denue/api/tests.py]: Better name the file according to what it is tested. And isolate tests and instructions to run specifc tests.

File: denue/api/views.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
+
+# Create your views here.
```
Comment (maurogonzalez) [inline denue/api/views.py]: Is this file needed?

**Review by maurogonzalez — CHANGES_REQUESTED** [review]
I reconsidered and it's better to add portability at this point and instructions to run specific tests per PR.

File: denue/api/tests.py (hunk 1 above)
Comment (jmercadoc) [reply denue/api/tests.py]: ok, i created tests files for each model 
commit: 4b7030b65b8ec05de1a2f6045937a382d4ea8765

File: denue/api/views.py (hunk 2 above)
Comment (jmercadoc) [reply denue/api/views.py]: no, this file was automatically created, it has beed deleted  
commit: d455945648dd7f1cd703b07ea2a1254da4e67e6a

File: .env.example [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+RENUE_API_KEY=your-denue-api=key
+DATABASE_ENGINE=django.contrib.gis.db.backends.postgis
+DATABASE_HOST=localhost
+DATABASE_NAME=postgres
+DATABASE_USERNAME=postgres
+DATABASE_PASSWORD=your-db-password
+DATABASE_PORT=5432
+DJANGO_DEBUG=True
+GDAL_LIBRARY_PATH=your-gdal-path
+GEOS_LIBRARY_PATH=your-geos-path
```
Comment (EmmaMireyaSm) [inline .env.example]: Is this `.env` file available in any place? I tried to run the tests and env variables are needed

File: README.md [hunk 4]
```diff
@@ -20,78 +23,67 @@ query them based on their location.
...
+$ python manage.py makemigrations api
+
+// Updates database schema
+$ python manage.py migrate api
+
+// Create virtual environment
+$ python -m venv env
+
+// Activate Virtual Environment
+$ source mypython/bin/activate
```
Comment (EmmaMireyaSm) [inline README.md]: The name of virtual environment is not the same

File: .env.example (hunk 3 above)
Comment (jmercadoc) [reply .env.example]: you should create it on the root directory of project

File: README.md (hunk 4 above)
Comment (maurogonzalez) [reply README.md]: @jmercadoc Better use docker instead of using venv locally, this can help to change local environments

Comment (jmercadoc) [reply README.md]: ok, I'll switch to docker

Comment (jmercadoc) [reply README.md]: thanks @EmmaMireyaSm , i fixe, commit 58fa8f395901a68d4814ed7f5346586bd1d90b02

File: .env.example (hunk 3 above)
Comment (EmmaMireyaSm) [reply .env.example]: Thanks for the `.env` file, I already used it. 
When I run `python manage.py test` the following exception is raised 
```
raise ImproperlyConfigured("Error loading psycopg2 module: %s" % e)
django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: No module named 'psycopg2'
```
I tried to install `psycopg2-binary` but another error raised during installation `error: command 'gcc' failed with exit status 1`

Comment (jmercadoc) [reply .env.example]: hi @EmmaMireyaSm  

change the way you work so as not to install dependencies on your computers, now I use Docker and add the commands to the README

thanks

**Review by maurogonzalez — COMMENTED** [review]
You can use the Dockerfile instead of building the image and the running the docker-compose after a change or at the beggining

**Review by maurogonzalez — APPROVED** [review]
Looks good!

File: .env.example (hunk 3 above)
Comment (maurogonzalez) [reply .env.example]: @jmercadoc to run what you want in this step for example :
```
docker-compose exec denue-api python denue/manage.py test api.tests.tests_model_comercial_activity

```

You don't really need to have the app running, right? Testing while modifying code can be easier if you use the code instead of a previously built image. What do you think?
